packages/babel-plugin-transform-es2015-parameters/src/default.js

```diff
@@ -44,17 +54,53 @@ const iifeVisitor = {
 };
 
 export const visitor = {
-  Function(path) {
+  Function(path, state) {
     const { node, scope } = path;
     if (!hasDefaults(node)) return;
 
     // ensure it's a block, useful for arrow functions
     path.ensureBlock();
 
-    const state = {
-      iife: false,
-      scope: scope,
-    };
+    const params = path.get("params");
+
+    if (state.opts.loose) {
+      const body = [];
+      for (let i = 0; i < params.length; ++i) {
+        const param = params[i];
+        if (param.isAssignmentPattern()) {
+
+          const left = param.get("left");
+          const right = param.get("right");
+
+          const undefinedNode = scope.buildUndefinedNode();
+
+          if ( left.isIdentifier() ) {
+            body.push(buildLooseDefaultParam({
+              ASSIGMENT_IDENTIFIER: left.node,
+              DEFAULT_VALUE: right.node,
+              UNDEFINED: undefinedNode,
+            }));
+            param.replaceWith(left.node);
+          }
+
+          if ( left.isObjectPattern() || left.isArrayPattern() ) {
```

Comment:
I cant think of any other cases either, but was not 100% sure so I made it explicit. I looked at the spec now and it seems to confirm this. I would still prefer the explicit version to make it clear why a different template is used in this case.